bugbug/tools/code_review_generation.py

```diff
@@ -0,0 +1,754 @@
+                                        if chunk.choices[0].delta.content is not None:
+                                            qualitative_feedback += chunk.choices[
+                                                0
+                                            ].delta.content
+
+                                if model == "claude-3-5-sonnet":
+                                    qualitative_feedback = client.messages.create(
+                                        model="claude-3-5-sonnet-20240620",
+                                        temperature=0.2,
+                                        max_tokens=10000,
+                                        system="You are a bot that provides qualitative feedback for a generated fix for a code review comment.",
+                                        messages=[
+                                            {
+                                                "role": "user",
+                                                "content": [
+                                                    {
+                                                        "type": "text",
+                                                        "text": feedback_prompt,
+                                                    }
+                                                ],
+                                            }
+                                        ],
+                                    )
+
+                                if metrics is not None:
+                                    writer.writerow(
+                                        [
+                                            revision_id,
+                                            patch_id,
+                                            prompt_type,
+                                            diff_length_limit,
+                                            hunk_size,
+                                            comment_length,
+                                            generated_code_length,
+                                            metrics["precision"],
+                                            metrics["recall"],
+                                            metrics["f1"],
+                                            qualitative_feedback,
+                                            file_path,
+                                            comment.content,
+                                            comment.start_line,
+                                            comment.end_line,
+                                            relevant_diff,
+                                            generated_fix,
+                                            reference_fix,
+                                        ]
+                                    )
+
+                                counter += 1
+
+                    else:
+                        print(f"No relevant diff found for Comment ID {comment.id}.\n")
```

Comment:
The evaluation code should also be moved into a separate script

Reply:
Separated here: 8873dacba5332c58db1e7728784b68f76cd33a04